### DISCUSSION TITLE
NextAuth on vercel returns 404 error, goes to url/api/auth/error but it works locally

### DISCUSSION THREAD

**Discussion by rohit4242** [nextauthjs/next-auth#7710] [Help]
### Question 💬

import { db } from "@/lib/db";
import { PrismaAdapter } from "@next-auth/prisma-adapter";
import { NextAuthOptions } from "next-auth";
import GoogleProvider from "next-auth/providers/google";

function getGoogleCredentials(): { clientId: string; clientSecret: string } {
  const clientId = process.env.GOOGLE_CLIENT_ID;
  const clientSecret = process.env.GOOGLE_CLIENT_SECRET;
  if (!clientId || clientId.length === 0) {
    throw new Error("Missing GOOGLE_CLIENT_ID");
  }

  if (!clientSecret || clientSecret.length === 0) {
    throw new Error("Missing GOOGLE_CLIENT_SECRET");
  }

  return { clientId, clientSecret };
}

export const authOptions: NextAuthOptions = {
  adapter: PrismaAdapter(db),
  session: {
    strategy: "jwt",
  },
  pages: {
    signIn: "/login",
  },
  providers: [
    GoogleProvider({
      clientId: getGoogleCredentials().clientId,
      clientSecret: getGoogleCredentials().clientSecret,
    }),
  ],
  secret: process.env.NEXTAUTH_SECRET,
  callbacks: {
    async session({ token, session }) {
      if (token) {
        session.user.id = token.id;
        session.user.name = token.name;
        session.user.email = token.email;
        session.user.image = token.picture;
      }

      return session;
    },
    async jwt({ token, user }) {
      const dbUser = await db.user.findFirst({
        where: {
          email: token.email,
        },
      });

      if (!dbUser) {
        token.id = user!.id;
        return token;
      }

      return {
        id: dbUser.id,
        name: dbUser.name,
        email: dbUser.email,
        picture: dbUser.image,
      };
    },
    redirect() {
      return "/dashboard";
    },
  },
};

### How to reproduce ☕️

i dont know

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by daniel21dev** [comment]
Verify your github clientId and secret, I had the same problem, my tokens were incorrect, I fixed it and then the auth worked.

**Comment by thenna** [comment]
same issue - I got the callback with the code /api/auth/callback/azure-ad?code=xxxxx but the page throws 404

**Comment by bn1ck** [comment]
When running locally instead of `npm run dev` run `vercel dev`. If the former works and the latter doesn't, something within your vercel.json is most likely misconfigured. In my case i completely removed it (the configuration i had in there wasn't needed)  and it worked.
